build_ngtf.py

```diff
@@ -344,79 +358,127 @@ def main():
+                "-DENABLE_SPEECH_DEMO=FALSE",
+                "-DCMAKE_INSTALL_RPATH=\"$ORIGIN\"",
+                "-DCMAKE_INSTALL_PREFIX=" + os.path.join(
+                    artifacts_location, "openvino")
+            ]
+
+            if arguments.debug_build:
+                openvino_cmake_flags.extend(["-DCMAKE_BUILD_TYPE=Debug"])
+
+            cmake_build(build_dir, openvino_src_dir, openvino_cmake_flags,
+                        verbosity)
+    else:
+        # Skip building nGraph if we're building OpenVINO
+        # Build nGraph if required.
+        if not arguments.use_prebuilt_ngraph:
+            ngraph_src_dir = './ngraph'
+            if arguments.ngraph_src_dir:
+                ngraph_src_dir = arguments.ngraph_src_dir
+                print("Using local nGraph source in directory ", ngraph_src_dir)
+            else:
+                if arguments.ngraph_version:
+                    ngraph_version = arguments.ngraph_version
+
+                print("nGraph Version: ", ngraph_version)
+                download_repo("ngraph",
+                              "https://github.com/NervanaSystems/ngraph.git",
+                              ngraph_version)
+
+            # Now build nGraph
+            ngraph_cmake_flags = [
+                "-DNGRAPH_INSTALL_PREFIX=" + artifacts_location,
+                "-DNGRAPH_USE_CXX_ABI=" + cxx_abi, "-DNGRAPH_DEX_ONLY=TRUE",
+                "-DNGRAPH_DEBUG_ENABLE=NO", "-DNGRAPH_UNIT_TEST_ENABLE=NO",
+                "-DNGRAPH_TARGET_ARCH=" + target_arch,
+                "-DNGRAPH_TUNE_ARCH=" + target_arch, "-DNGRAPH_TBB_ENABLE=FALSE"
+            ]
+
+            if arguments.use_ngraph_staticlibs:
+                ngraph_cmake_flags.extend(["-DNGRAPH_STATIC_LIB_ENABLE=TRUE"])
+                ngraph_cmake_flags.extend(
+                    ["-DNGRAPH_CPU_STATIC_LIB_ENABLE=TRUE"])
+                ngraph_cmake_flags.extend(
+                    ["-DNGRAPH_INTERPRETER_STATIC_LIB_ENABLE=TRUE"])
+                ngraph_cmake_flags.extend(
+                    ["-DNGRAPH_DYNAMIC_COMPONENTS_ENABLE=OFF"])
+
+            if arguments.debug_build:
+                ngraph_cmake_flags.extend(["-DCMAKE_BUILD_TYPE=Debug"])
+
+            if arguments.build_plaidml_backend:
+                command_executor(["pip", "install", "-U", "plaidML"])
+
+            ngraph_cmake_flags.extend([
+                "-DNGRAPH_TOOLS_ENABLE=" +
+                flag_string_map[platform.system() != 'Darwin']
+            ])
+            ngraph_cmake_flags.extend([
+                "-DNGRAPH_GPU_ENABLE=" +
+                flag_string_map[arguments.build_gpu_backend]
+            ])
+            ngraph_cmake_flags.extend([
+                "-DNGRAPH_PLAIDML_ENABLE=" +
+                flag_string_map[arguments.build_plaidml_backend]
+            ])
+            ngraph_cmake_flags.extend([
+                "-DNGRAPH_INTELGPU_ENABLE=" +
+                flag_string_map[arguments.build_intelgpu_backend]
+            ])
```

Comment:
As discussed, we can remove the backend specific flags/tests/code in a separate PR. Commenting to keep a note.